lib/src/layer/tile_layer/tile_renderer.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flutter_map/src/layer/tile_layer/tile_coordinates.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_image.dart';
+
+/// Display of a [TileImage] at given [TileCoordinates].
+///
+/// In most cases, the [positionCoordinates] are equal to tileImage coordinates.
+/// Except when we display several worlds in the same map, or when we cross the
+/// 180/-180 border.
+class TileRenderer {
+  /// TileImage to display.
+  final TileImage tileImage;
+
+  /// Position where to display [tileImage].
+  final TileCoordinates positionCoordinates;
+
+  /// Create an instance of [TileRenderer].
+  const TileRenderer(this.tileImage, this.positionCoordinates);
+
+  @override
+  bool operator ==(Object other) {
+    // TODO may not be good enough with the ObjectKey
+    return other is TileRenderer &&
+        other.positionCoordinates == positionCoordinates;
+  }
```

Comment:
Include `identical` before the more specific checks, and double check this todo.

Reply:
I've added the `identical` test.
I've also double-checked the theoretical usage of `ObjectKey`/`identical`: `TileRenderer`s with different `TileImage`s will be considered as different. The `ObjectKey` will differ - as it did before this PR - with a different `TileImage` object.